Create Jupyter Notebooks post (#1137)
agrc/gis.utah.gov: _posts/2019-06-25-using-jupyter-notebook-for-agrc-collaboration.md

---
title: 'Using Jupyter Notebook for AGRC Collaboration'
author:
  display_name: Keaton Walker
  email: kwalker@utah.gov
date: 2019-06-25 21:29:10
categories:

>   - Featured

Reviewer: ```suggestion
  - Featured
  - Developer
```

tags:
  - notebook
  - jupyter

>   - pyton

Reviewer: ```suggestion
  - python
```

One of the things we recently decided to track usage metrics on is our SDE SQL server, which  is full of GIS tables. Most of our users know this database as sgid.agrc.utah.gov. Tracking usage of a database can be done in many ways: we could track database connections, transactions per second, or even CPU usage. But we wanted a metric that captured some of the unique aspects of using our database with ArcGIS clients, so we chose to track shared table locks. (The SGID lock metrics are stored in BigQuery, so you’ll need a free GCP project to run this notebook yourself.) This [exploration of the SGID logs notebook](https://colab.research.google.com/drive/1kgm0GRyNR80tSeSJVnKQbZVXS_Ytr5S1) shows some of the information that can be gleaned from shared lock counts when they are aggregated by both application server machines and desktop computer users. 

![Shared lock counts]({{ "/images/DesktopLockCount.png" | prepend: site.baseurl }})

> Working with log data, or any other source of data, can involve multiple data transformations. Sometimes many transformations and analysis need to be applied to data, and the work is distributed across multiple people in our office. When more than one person is involved in a process, good communication and documentation become much more important. This [notebook created with the ArcGIS Pro environment](https://github.com/keatonwalker/multimodal/blob/master/gtfs_parsing/trip_stops.ipynb) shows work done for our current [multi-modal project](https://gis.utah.gov/multi-modal-travel-shed-analysis-project/). For this project, the general transit feed specification (GTFS) files produced by the Utah Transit Authority need to be transformed into a format that would work with ESRI’s Network Analyst while maintaining the ID structure of the original GTFS. The notebook shows how we parsed the GTFS files and transformed them into a shapefile format that could be consumed by ArcGIS. The notebook was created with the ArcGIS Pro installation of Python and was then pushed to Github. Github has a very nice feature of rendering notebooks with outputs and code shown in the Github web application.

Reviewer: links should never contain `gis.utah.gov`. An example https://github.com/agrc/gis.utah.gov#link-to-post

```suggestion
Working with log data, or any other source of data, can involve multiple data transformations. Sometimes many transformations and analysis need to be applied to data, and the work is distributed across multiple people in our office. When more than one person is involved in a process, good communication and documentation become much more important. This [notebook created with the ArcGIS Pro environment](https://github.com/keatonwalker/multimodal/blob/master/gtfs_parsing/trip_stops.ipynb) shows work done for our current [multi-modal project]({{site.baseurl}}{% post_url 2018-11-07-multi-modal-travel-shed-analysis-project %}). For this project, the general transit feed specification (GTFS) files produced by the Utah Transit Authority need to be transformed into a format that would work with ESRI’s Network Analyst while maintaining the ID structure of the original GTFS. The notebook shows how we parsed the GTFS files and transformed them into a shapefile format that could be consumed by ArcGIS. The notebook was created with the ArcGIS Pro installation of Python and was then pushed to Github. Github has a very nice feature of rendering notebooks with outputs and code shown in the Github web application.
```

Author: Sorry! Didn't really look too closely at the links Keaton included. Will make note to check them for internal posts next time!

Reviewer: not a problem. i figured that was the case.
